### PR TITLE
fixes #773

### LINKED ISSUE #773: [BUG] DlSmartSearch - when having aliased values of nested schema it does not hide them
image

Expect not to see metadata.system.height

image

### PR DESCRIPTION
**Pull request by makc-dataloop**

### CODE REVIEW

File: src/hooks/use-suggestions.ts [hunk 1]
```diff
@@ -272,12 +272,24 @@ export const useSuggestions = (
...
                         for (const key of Object.keys(fieldOf)) {
                             if (key === '*') continue
+                            if (aliasedKeys.includes(`${matchedField}.${key}`))
+                                continue
                             toAdd.push(`.${key}`)
                         }
                     }
                 } else {
+                    const matchedFieldBeforeDot = matchedField.replace(
+                        new RegExp(`\\.${lastWord}$`),
```
Comment (guyDataloop) [inline src/hooks/use-suggestions.ts:282]: The double backslash seems wrong, if you want to escape the `.` you need only `\.`

File: src/hooks/use-suggestions.ts [hunk 2]
```diff
@@ -272,12 +272,24 @@ export const useSuggestions = (
                     if (isObject(fieldOf) && !Array.isArray(fieldOf)) {
                         for (const key of Object.keys(fieldOf)) {
                             if (key === '*') continue
+                            if (aliasedKeys.includes(`${matchedField}.${key}`))
```
Comment (guyDataloop) [inline src/hooks/use-suggestions.ts:275]: How does it work when the attributes have mutual strings that they start with ? Would the suggestion be correct ? Maybe should regex it as a starter or have a point before the `matchedField`, and check if it;s the end after `key`.

For exmaple `aaa.bbb` and `aa.bb`

File: src/hooks/use-suggestions.ts [hunk 3]
```diff
@@ -272,12 +272,24 @@ export const useSuggestions = (
...
                 } else {
+                    const matchedFieldBeforeDot = matchedField.replace(
+                        new RegExp(`\\.${lastWord}$`),
+                        ''
+                    )
                     for (const key in fieldOf) {
                         if (key === '*') continue
+                        if (
+                            aliasedKeys.includes(
+                                `${matchedFieldBeforeDot}.${key}`
```
Comment (guyDataloop) [inline src/hooks/use-suggestions.ts:289]: Same

**Review by guyDataloop — CHANGES_REQUESTED** [review]

File: src/hooks/use-suggestions.ts (hunk 1 above)
Comment (makc-dataloop) [reply src/hooks/use-suggestions.ts:282]: I actually want to escape slash itself, so the resulting string would have \.

File: src/hooks/use-suggestions.ts (hunk 2 above)
Comment (makc-dataloop) [reply src/hooks/use-suggestions.ts:275]: this does not change how it works, it just skips the keys for which we have aliases. if there is no alias, the condition fails and the code proceeds as before

**Review by guyDataloop — APPROVED** [review]

**Pull request merged**